FWCore/Framework/test/test_catch2notTP_InputProcessBlock.cc

```diff
@@ -0,0 +1,68 @@
+
+  class TestProducerWithoutInputProcessBlockAbility : public edm::stream::EDProducer<edm::GlobalCache<int>> {};
+
+}  // namespace edmtest
+
+TEST_CASE("test InputProcessBlock", "[InputProcessBlock]") {
+  SECTION("test HasAbility") {
+    REQUIRE(edmtest::TestAnalyzerWithInputProcessBlockAbility::HasAbility::kInputProcessBlockCache);
+    REQUIRE(!edmtest::TestAnalyzerWithoutInputProcessBlockAbility::HasAbility::kInputProcessBlockCache);
+  }
+  SECTION("test type aliases") {
+    REQUIRE(std::is_same<edmtest::TestAnalyzerWithInputProcessBlockAbility::InputProcessBlockCache,
+                         edm::impl::InputProcessBlockCacheImpl<int, unsigned int, int>>());
+    REQUIRE(std::is_same<edmtest::TestAnalyzerWithoutInputProcessBlockAbility::InputProcessBlockCache, void>());
+    REQUIRE(std::is_same<edm::stream::impl::choose_unique_ptr<
+                             edmtest::TestAnalyzerWithInputProcessBlockAbility::InputProcessBlockCache>::type,
+                         std::unique_ptr<edm::impl::InputProcessBlockCacheImpl<int, unsigned int, int>>>());
+    REQUIRE(std::is_same<edm::stream::impl::choose_unique_ptr<
+                             edmtest::TestAnalyzerWithoutInputProcessBlockAbility::InputProcessBlockCache>::type,
+                         edm::stream::impl::dummy_ptr>());
+
+    REQUIRE(std::is_same<edmtest::TestFilterWithInputProcessBlockAbility::InputProcessBlockCache,
+                         edm::impl::InputProcessBlockCacheImpl<int, unsigned int, int>>());
+    REQUIRE(std::is_same<edmtest::TestFilterWithoutInputProcessBlockAbility::InputProcessBlockCache, void>());
+    REQUIRE(std::is_same<edm::stream::impl::choose_unique_ptr<
+                             edmtest::TestFilterWithInputProcessBlockAbility::InputProcessBlockCache>::type,
+                         std::unique_ptr<edm::impl::InputProcessBlockCacheImpl<int, unsigned int, int>>>());
+    REQUIRE(std::is_same<edm::stream::impl::choose_unique_ptr<
+                             edmtest::TestFilterWithoutInputProcessBlockAbility::InputProcessBlockCache>::type,
+                         edm::stream::impl::dummy_ptr>());
+
+    REQUIRE(std::is_same<edmtest::TestProducerWithInputProcessBlockAbility::InputProcessBlockCache,
+                         edm::impl::InputProcessBlockCacheImpl<int, unsigned int, int>>());
+    REQUIRE(std::is_same<edmtest::TestProducerWithoutInputProcessBlockAbility::InputProcessBlockCache, void>());
+    REQUIRE(std::is_same<edm::stream::impl::choose_unique_ptr<
+                             edmtest::TestProducerWithInputProcessBlockAbility::InputProcessBlockCache>::type,
+                         std::unique_ptr<edm::impl::InputProcessBlockCacheImpl<int, unsigned int, int>>>());
+    REQUIRE(std::is_same<edm::stream::impl::choose_unique_ptr<
+                             edmtest::TestProducerWithoutInputProcessBlockAbility::InputProcessBlockCache>::type,
+                         edm::stream::impl::dummy_ptr>());
```

Comment:
All these could be `static_assert`s, right? A quick search pointed out that Catch2 2.4.2 included `STATIC_REQUIRE`, which works like `static_assert`, but records the success at runtime as well
https://github.com/catchorg/Catch2/blob/devel/docs/other-macros.md

Our Catch2 version appears to be 2.2.2, maybe it is time to ask it to be updated (I did in https://github.com/cms-sw/cmsdist/issues/7052).

Reply:
Done. I changed them to use STATIC_REQUIRE. That should be better.